requirements.txt

```diff
@@ -36,17 +36,17 @@ names-generator==0.1.0
 numpy==1.25.0; python_version >= '3.9'
 packaging==23.1; python_version >= '3.7'
 parse==1.19.1; python_version >= '3.0'
-parse-type==0.6.1; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
+parse-type==0.6.2; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
 pathspec==0.11.1; python_version >= '3.7'
 pillow==10.0.0; python_version >= '3.8'
-platformdirs==3.8.0; python_version >= '3.7'
+platformdirs==3.8.1; python_version >= '3.7'
 pluggy==1.2.0; python_version >= '3.7'
 pretend==1.0.9
 py==1.11.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'
 pycodestyle==2.10.0; python_version >= '3.6'
 pyflakes==3.0.1; python_version >= '3.6'
 pygments==2.15.1; python_version >= '3.7'
-pyparsing==3.1.0; python_full_version >= '3.6.8'
```

Comment:
Why would dependabot downgrade a library version?

Reply:
It would happen when two dependencies use pyparsing, but one is incompatible with 3.1.0 version.
`Pipenv` handles it.
I see it in `pipenv graph`

```shell
sphinxcontrib-needs==0.7.9
  - jsonschema [required: ==3.2.0, installed: 3.2.0]
    - attrs [required: >=17.4.0, installed: 23.1.0]
    - pyrsistent [required: >=0.14.0, installed: 0.19.3]
    - setuptools [required: Any, installed: 68.0.0]
    - six [required: >=1.11.0, installed: 1.16.0]
  - matplotlib [required: >3.1.0, installed: 3.7.2]
    - contourpy [required: >=1.0.1, installed: 1.1.0]
      - numpy [required: >=1.16, installed: 1.25.1]
    - cycler [required: >=0.10, installed: 0.11.0]
    - fonttools [required: >=4.22.0, installed: 4.40.0]
    - kiwisolver [required: >=1.0.1, installed: 1.4.4]
    - numpy [required: >=1.20, installed: 1.25.1]
    - packaging [required: >=20.0, installed: 23.1]
    - pillow [required: >=6.2.0, installed: 10.0.0]
    - pyparsing [required: >=2.3.1,<3.1, installed: 3.0.9]   # <---
```